todolist/todo/api.py

```diff
@@ -1,23 +1,120 @@
 from django.contrib.auth.models import User
 from tastypie import fields
-from tastypie.resources import ModelResource
+from tastypie.resources import ModelResource, Resource, ALL, ALL_WITH_RELATIONS
 from todo.models import Task
 from tastypie.authentication import SessionAuthentication
+from tastypie.authorization import DjangoAuthorization,Authorization
+from django.contrib.auth import authenticate, login, logout
+from tastypie.http import HttpUnauthorized, HttpForbidden
+from django.conf.urls import  url
+from tastypie.utils import trailing_slash
+from django.core.exceptions import ObjectDoesNotExist
+from authorization import UsersAuthorization, TasksAuthorization
 
 
+###################User resource####################
+
 class UserResource(ModelResource):
     class Meta:
         queryset = User.objects.all()
-        resource_name = 'users'
+        resource_name = 'user'
+        allowed_methods = ['get', 'post']
         authentication = SessionAuthentication()
-        #fields = ['username', 'first_name', 'last_name', 'last_login']
+        authorization=UsersAuthorization()
+        excludes = ['email', 'password', 'is_active', 'is_staff',
+                    'is_superuser','last_login','last_name',
+                    'date_joined','first_name']
         #allowed_methods = ['get']
+        filtering = {
+            'username': ALL,
+        }
 
+###################Task resource####################
 
 class TaskResource(ModelResource):
-    user = fields.ForeignKey(UserResource, 'user')
+  user = fields.ForeignKey(UserResource, 'user')
 
-    class Meta:
-        queryset = Task.objects.all()
-        resource_name = 'tasks'
-        #excludes = ['email', 'password', 'is_active', 'is_staff', 'is_superuser']
+  class Meta:
+    authorization = TasksAuthorization() #Authorization()
+    queryset = Task.objects.all()
+    allowed_methods = ['get', 'post', 'put','patch','delete']
+    resource_name = 'tasks'
+    always_return_data=True
+    filtering = {
+            'user': ALL_WITH_RELATIONS,
+        }
+
+
+###################Login resource####################
+
+class LoginResource(Resource):
+  class Meta:
+    resource_name = 'login'
+
+  def dispatch(self, request_type, request, **kwargs):
+    self.method_check(request, allowed=['post'])
+    data = self.deserialize(request, request.body,
+                            format=request.META.get('CONTENT_TYPE', 'application/json'))
+    username = data.get('username', '')
```

Comment:
do you have concerns regarding `request.POST.get('password', '')`? it looks easier